### ISSUE TITLE
Every time a xdg_popup is created, there is a wlroots error log

### ISSUE THREAD

**Issue by lilydjwg** [WayfireWM/wayfire#2189]
**Describe the bug**

**To Reproduce**
1. Create a xdg_popup somehow (e.g. context menus)
2. Check the log:

```
II 09-03-24 23:00:58.124 - [build/wayfire/src/view/xdg-shell.cpp:59] New xdg popup
EE 09-03-24 23:00:58.124 - [types/xdg_shell/wlr_xdg_surface.c:169] A configure is scheduled for an uninitialized xdg_surface 0x557ffd217530
```

**Expected behavior**
No error logs.

**Wayfire version**
git.

This patch fixes it, but I don't know if it introduces any issue:

```diff
diff --git a/src/view/xdg-shell.cpp b/src/view/xdg-shell.cpp
index 8340e932..09747202 100644
--- a/src/view/xdg-shell.cpp
+++ b/src/view/xdg-shell.cpp
@@ -110,7 +110,6 @@ std::shared_ptr<wayfire_xdg_popup> wayfire_xdg_popup::create(wlr_xdg_popup *popu
     self->surface_root_node = std::make_shared<wayfire_xdg_popup_node>(self->get_id());
     self->set_surface_root_node(self->surface_root_node);
     self->set_output(self->popup_parent->get_output());
-    self->unconstrain();
     return self;
 }
 
```

**Comment by ammen99** [comment]
Yeah it should be safe to remove this call. It should happen on map() instead (if it isn't happening already)

**Comment by lilydjwg** [comment]
I've reverted my local change because I suspend it caused my Firefox extension's popup to misplace. I may need to do some debugging.

**Comment by lilydjwg** [comment]
> It should happen on map() instead (if it isn't happening already)

I tried to move it there but the position is still wrong.

To reproduce, maximize Firefox, put an extension button on the left and click. Part of it will be out of screen:

![图片](https://github.com/WayfireWM/wayfire/assets/440661/be9a3455-66a7-48b4-8063-c5dbb674ceb9)

**Comment by ammen99** [comment]
Sway does this: https://github.com/swaywm/sway/blob/c2f08075ec00632293bbc63582c7f3ffd75441af/sway/desktop/xdg_shell.c#L67-L72

We already have a listener for the commit, but we connect it on map() only. I suppose we should connect at creation time, unconstrain on first commit, and the rest if the popup is mapped https://github.com/WayfireWM/wayfire/blob/99e61eb7cd3dc47c5a27efd1ff45f35611e337ba/src/view/xdg-shell.cpp#L238-L242

If you want, try it and see how well it works ;)
